### PR TITLE
experimental: remove gulp from appium package

### PR DESCRIPTION
**Pull request by boneskull**
- Replaced gulp with babel, mocha, and npm scripts
    - Some redundancy necessary to support running scripts from monorepo root or `packages/appium`
    - `build/lib` => `build` now that we do not compile test files and have moved scripts from `commands-yml` to `scripts`. This may be a breaking change
- Moved `commands-yml/*.js` to scripts under `packages/appium/scripts/`.
- Refactored all scripts to be CJS, so they do not need transpilation.  Alternatively, we could have renamed them all `*.mjs` and used ESM, but most were CJS already.  For now, ESLint will disallow use of `import` keyword in this script folder for `.js` files.
- Normalized test file names to align with common convention (some IDEs now recognize the files as test files and behave accordingly)
    - `*-e2e-specs.js` => `*.e2e.spec.js`
    - `*-specs.js` => `*.spec.js`
- Reorganized tests to be more easily executed on CLI (E2E tests in `e2e`; all else in `unit`). Moved a fixture or two
- Sorted `package.json`s
- Moved `zip`/`upload`/`zip-and-upload` scripts to root (see "problems" below)
- Added `@babel/cli` dev dep
- Updated `.eslintignore`, `.eslintrc`, etc.
- Mocha config: added a check to ensure `.only()` doesn't get committed, sneak in to CI and give false positives

## Problems

- XXX: The `upload` script does not work, because there is no gulpfile here.  Is the `github-upload` task something that should be a script?  Do we use it elsewhere?
- ~~The `dev` script (and `watch` script) in the monorepo root `package.json` do not work, since we'd need another program (e.g., `concurrently`) to run `gulp --watch` and `babel --watch` in parallel.  When gulp is gone, this will get fixed.  For now, launch `dev` and the new `dev:appium` in separate terminals~~. 
**UPDATE:** This now works properly.

### CODE REVIEW

**Comment by boneskull** [comment]
Current dependencies on/for this PR:
* 2.0
  * **PR #16443** <a href="https://app.graphite.dev/github/pr/appium/appium/16443" target="_blank">Graphite</a> 
    * **PR #16482** <a href="https://app.graphite.dev/github/pr/appium/appium/16482" target="_blank">Graphite</a> 
    * **PR #16459** <a href="https://app.graphite.dev/github/pr/appium/appium/16459" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Comment by boneskull** [comment]
@jlipps Looking for guidance on the "github upload" script.  Do we still need it?  If we don't, I'll move this out of draft.

File: packages/appium/scripts/parse-yml-commands.js [hunk 1]
```diff
@@ -1,13 +1,92 @@
...
+};
+
+const CLIENT_URL_TYPES = {
+  url: 'hostname',
+  android: 'Android',
+  ios: 'iOS',
+};
+
+const validator = {
+  'name': {presence: true},
```
[the remark this answers is not in the document]
Comment (jlipps) [reply packages/appium/scripts/parse-yml-commands.js:65]: though i get doing it here for consistency might look nicer

**Review by jlipps — APPROVED** [review]
looks good, however it doesn't seem to remove gulp? maybe i misunderstood the point

File: package.json [hunk 2]
```diff
@@ -24,44 +24,49 @@
   "license": "Apache-2.0",
   "author": "https://github.com/appium",
   "scripts": {
-    "build": "lerna exec gulp transpile",
+    "build": "lerna exec --parallel -- gulp transpile",
```
[the remark this answers is not in the document]
Comment (boneskull) [reply package.json:27]: Good catch, I missed this.

File: packages/appium/scripts/parse-yml-commands.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (boneskull) [reply packages/appium/scripts/parse-yml-commands.js:65]: this is a copy-paste job

File: packages/appium/scripts/write-server-args-docs.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    '..',
+    '..',
+    '..',
+    'docs',
+    'en',
+    'writing-running-appium',
+    'server-args.md',
+  ),
+);
+let md = '# Appium server arguments\n\n';
```
[the remark this answers is not in the document]
Comment (boneskull) [reply packages/appium/scripts/write-server-args-docs.js]: yes, might as well.

**Comment by boneskull** [comment]
I'm going to revert the change which outputs the contents of `lib` into `build`; instead, the result will be under `build/lib` as it was before.  This would otherwise be a breaking change to [the java client](https://github.com/appium/java-client/blob/9fa0a13e2f1684bd996963b1f6bd1a962c879d21/src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java#L76) and [the python client](https://github.com/appium/python-client/blob/895cde1aa674aaab2f958ae251de0daefd049c02/appium/webdriver/appium_service.py#L26).  This should probably change (because `build` will _only_ contain `lib`), but it doesn't have to happen now.

**Pull request merged**